File: program/program/find-sum-of-squares-of-numbers-by-recursion/FindSumOfSquaresOfNumbersByRecursion.java
import java.util.Arrays;

class FindSumOfSquaresOfNumbersByRecursion {
    public static void main(String[] args) {
        String input = "1 2 3 4 5";
        int[] arr = Arrays.stream(input.split(" ")).mapToInt(Integer::parseInt).toArray();
        int n = arr.length;
        System.out.println("Result : " + sumOfSquares(arr, n)); // 55
    }

    private static int sumOfSquares(int[] arr, int n) {
        if (n == 0) {
            return 0;
        }
        return (arr[n - 1] * arr[n - 1]) + sumOfSquares(arr, n - 1);
    }
}

File: program/program/find-sum-of-squares-of-numbers-by-recursion/find_sum_of_squares_of_numbers_by_recursion.go
package main

import "fmt"

func sumSquareRecursive(arr []int, index int) int {
	if index < 0 {
		return 0
	}
	return arr[index] * arr[index] + sumSquareRecursive(arr, index - 1)
}

func main() {
	input := []int{1, 2, 3, 4, 5}
	fmt.Println(sumSquareRecursive(input, len(input) - 1))
}
